tools/shared.py

```diff
@@ -1896,7 +1896,9 @@ def generate_string_initializer(s):
     s = s.replace('\\', '\\\\').replace("'", "\\'")
     s = s.replace('\n', '\\n').replace('\r', '\\r')
     def escape(x): return '\\x{:02x}'.format(ord(x.group()))
-    return re.sub('[\x80-\xff]', escape, s)
+    s = re.sub('\x00(?![0-9])', '\\0', s) # Don't allow a null byte in the generated output file.
+    s = re.sub('[\x00-\x1f\x80-\xff]', escape, s) # Don't allow ASCII chars 0-31 or 128-255 in the generated output file.
+    return s
```

Comment:
I remember that in https://github.com/kripken/emscripten/pull/3326#issuecomment-106743344 I detected someone postprocessing the strings, and in https://github.com/kripken/emscripten/pull/3326#issuecomment-110268477 I identified uglify as the culprit. So there was some setup where I could enable that processing step without even trying. Likely tied to some optimization level. Not sure whether that applies anymore; I've not been updating my emscripten toolbox in some time. If the code still gets run, it should affect Windows builds the same way as the original problem here.